.github/workflows/CD-pipeline-dev.yml

```diff
@@ -26,19 +26,21 @@ jobs:
         run: |
           SERVICES=( ${{ env.SERVICES }} )
           git fetch origin master
-          echo "Files changed since last commit"
-          git diff  --name-only HEAD^ HEAD
-          for i in "$(git diff  --name-only HEAD^ HEAD)"
+          changes=($(git diff  --name-only HEAD^ HEAD))
+          for i in "${changes[@]}"
           do
-            if [[ $i == *"internal/"* ]] || [[ $i == *"proto/"* ]] || [[ $i == "go.mod" ]] || [[ $i == "go.sum" ]] && [[ $i != *".md"* ]]; then
+            if [[ $i == *".md"* ]]; then
```

Comment:
I don't mean to pick or anything; could this perhaps be better future-proofed by quoting variables ("$stuff") to prevent possible word splitting scenarios? not sure if it's sufficient that the array being iterated over is quoted, but IMO it wouldn't hurt to quote all variables for bash..

Reply:
I think you wanted to do regexp comparison here, didn't you? By no means showing off here, but as we have issues with this CI, let's make it a bit more robust using the regexes:
```
if [[ "${i}" =~ .*\.md.* ]]; then
  ...
elif [[ "${i}" =~ .*(internal|proto|manifests)/.* ]] || [[ "${i}" =~ go\.(mod|sum) ]]; then
...
else
  for SERVICE in "${SERVICES[@]}"; do
    if [[ "${i}" =~ .*services/${SERVICE}.* ]]; then
...
```
Also, I've combined lines 35 and 38. You can take similar approach with the regexes on line

Reply:
By no means, please leave any feedback you think will add value. FIY @wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf 

Tbf, my regex skills are quite poor, so thanks for the examples/improvements, will definitely add it.

Reply:
looks fine to me now;

it'd be great if we could test the CI config with some kind of unit tests... :thinking: :laughing: